build.cake

```diff
@@ -104,39 +104,96 @@ Task("Build")
-               .WithProperty("TargetFramework", framework)
-               .WithProperty("PublishDir", BIN_DIR + "agents/" + framework));
+    foreach (var framework in new[] { "netcoreapp3.1", "net5.0" })
+        MSBuild(AGENT_PROJECT, CreateMSBuildSettings("Publish")
+            .WithProperty("TargetFramework", framework)
+            .WithProperty("PublishDir", BIN_DIR + "agents/" + framework));
 
-        foreach (var framework in new[] { "netstandard2.0" })
-            MSBuild(ENGINE_API_CSPROJ, CreateMSBuildSettings("Publish")
-               .WithProperty("TargetFramework", framework)
-               .WithProperty("PublishDir", BIN_DIR + framework));
+    foreach (var framework in new[] { "netstandard2.0" })
+        MSBuild(ENGINE_API_PROJECT, CreateMSBuildSettings("Publish")
+            .WithProperty("TargetFramework", framework)
+            .WithProperty("PublishDir", BIN_DIR + framework));
 
-        foreach (var framework in new [] { "netcoreapp2.1", "netcoreapp3.1" })
-             MSBuild(ENGINE_TESTS_CSPROJ, CreateMSBuildSettings("Publish")
-                .WithProperty("TargetFramework", framework)
-                .WithProperty("PublishDir", BIN_DIR + framework));
+    foreach (var framework in new[] { "netcoreapp2.1", "netcoreapp3.1", "net5.0" })
+        MSBuild(ENGINE_TESTS_PROJECT, CreateMSBuildSettings("Publish")
+            .WithProperty("TargetFramework", framework)
+            .WithProperty("PublishDir", BIN_DIR + framework));
 
-        MSBuild(CONSOLE_CSPROJ, CreateMSBuildSettings("Publish")
-            .WithProperty("TargetFramework", "netcoreapp3.1")
-            .WithProperty("PublishDir", BIN_DIR + "netcoreapp3.1"));
+    MSBuild(CONSOLE_PROJECT, CreateMSBuildSettings("Publish")
+        .WithProperty("TargetFramework", "netcoreapp3.1")
+        .WithProperty("PublishDir", BIN_DIR + "netcoreapp3.1"));
 
-         MSBuild(CONSOLE_TESTS_CSPROJ, CreateMSBuildSettings("Publish")
-            .WithProperty("TargetFramework", "netcoreapp3.1")
-            .WithProperty("PublishDir", BIN_DIR + "netcoreapp3.1"));
+    MSBuild(CONSOLE_TESTS_PROJECT, CreateMSBuildSettings("Publish")
+        .WithProperty("TargetFramework", "netcoreapp3.1")
+        .WithProperty("PublishDir", BIN_DIR + "netcoreapp3.1"));
+}
 
-    });
+private void BuildEachProjectSeparatelyOnLinux()
+{
+    DotNetRestore(SOLUTION_FILE);
+
+    BuildProject(ENGINE_PROJECT, "net20", "netstandard2.0", "netcoreapp3.1");
+    BuildProject(CONSOLE_PROJECT, "net20", "netcoreapp3.1");
+    BuildProject(AGENT_PROJECT, "net20", "net40", "netcoreapp3.1", "net5.0");
+    BuildProject(AGENT_X86_PROJECT, "net20", "net40");
+
+    BuildProject(ENGINE_TESTS_PROJECT, "net35", "netcoreapp2.1", "netcoreapp3.1", "net5.0");
+    BuildProject(CONSOLE_TESTS_PROJECT, "net35", "netcoreapp3.1");
+
+    BuildProject(MOCK_ASSEMBLY_X86_PROJECT, "net35", "net40", "netcoreapp2.1", "netcoreapp3.1");
+    BuildProject(NOTEST_PROJECT, "net35", "netcoreapp2.1", "netcoreapp3.1");
```

Comment:
How come I cannot see this for windows/macOS ?

Reply:
Because we build the solution for Windows and MacOS, which we cannot do for Linux. I spent a lot of time trying to make building the solution work, but was defeated by the net35 projects. In fact, I think fewer stuff will work under a .NET Core build as time goes on and other runtimes go out of support.